feature_engine/transformation/date_transformer.py

```diff
@@ -0,0 +1,49 @@
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine.base_transformers import BaseNumericalTransformer
+from feature_engine.variable_manipulation import _check_input_parameter_variables
+
+
+class date_feature_extractor(BaseNumericalTransformer):
+    def __init__(
+        self,
+        features: Union[None, int, str, List[Union[str, int]]] = None,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+    ) -> None:
+
+        if not fearures:
+            raise ValueError(
+                "Feature list cannot be empty.Enter features to extract from dates"
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.fearures = set(fearures)
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+
+        X = super().fit(X)
+
+        # check if all the variables are of type datetime.
+        if not all([pd.is_datetime64_any_dtype(X[col]) for col in X.columns]):
```

Comment:
is this the only datetime format allowed / understood by pandas?

Reply:
Also, it should check only the type of self.variables and not X.columns. We only want those to be transformed to be datetime